### PR TITLE
Introduce Typescript

### PR DESCRIPTION
**Pull request by dbrgn**
- Add .editorconfig
- Upgrade telegram bot library (this upgrades to Telegram Bot API version 3)
- Add TypeScript

Everything can be compiled with `tsc` and seems to work right now.

I want to do further improvements, including a "watch" mode for compilation.

@netzhuffle do you want the setup to generate a "dist" js bundle (which can be committed when releasing a new version), or is running "tsc" before running the code fine?

### CODE REVIEW

File: src/FlameBot.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        } else {
+          // CYBER
+          sticker = new Sticker('CAADAgADlgADGW8XB1uhplTaDTPOAg');
+        }
+        this.reply(sticker, message);
+        return;
+      }
+
+      // Mentions
+      const botUsername = await this.usernamePromise;
```
Comment (dbrgn) [inline src/FlameBot.ts:124]: We're living in the future :grin:

**Comment by dbrgn** [comment]
I think this is ready for review / merge now.

@netzhuffle @rnestler want to review?

@netzhuffle if you want a "dist" bundle, that can be added using a separate PR (e.g. using webpack).

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+[*]
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
+charset = utf-8
+
+[*.{js,ts,json}]
+indent_style = space
+indent_size = 2
```
Comment (dbrgn) [inline .editorconfig:13]: I prefer 4 spaces, but set it to 2 to match the current style.

**Comment by dbrgn** [comment]
Now with CI :slightly_smiling_face:

**Comment by dbrgn** [comment]
@rnestler @netzhuffle ping

File: .editorconfig (hunk 2 above)
Comment (rnestler) [reply .editorconfig:13]: Since it is almost a complete rewrite anyway I wouldn't mind if you'd change it to 4.

**Comment by dbrgn** [comment]
Merge due to inactivity and no objections.

**Pull request merged**